api/src/main/java/io/kafbat/ui/controller/BrokersController.java

```diff
@@ -94,10 +97,24 @@ public Mono<ResponseEntity<Flux<BrokerConfigDTO>>> getBrokerConfig(String cluste
         .operationParams(Map.of(BROKER_ID, id))
         .build();
 
+    Flux<BrokerConfigDTO> brokerConfigs = brokerService.getBrokerConfig(getCluster(clusterName), id)
+        .map(clusterMapper::toBrokerConfig);
+
+    var kafkaCluster = clustersStorage.getClusterByName(clusterName)
+        .orElseThrow(
+            () -> new ClusterNotFoundException(
+                String.format("No cluster for name '%s'", clusterName)));
+
+    if (kafkaCluster.isReadOnly()) {
+      brokerConfigs = brokerConfigs.map(config -> {
+        config.setIsReadOnly(true);
+        return config;
+      });
+    }
+
     return validateAccess(context).thenReturn(
         ResponseEntity.ok(
-            brokerService.getBrokerConfig(getCluster(clusterName), id)
-                .map(clusterMapper::toBrokerConfig))
+            brokerConfigs)
```

Comment:
we can get the cluster earlier for a var, but let's call getConfigs only after validateAccess, I see two reasons for this:
-  we might not have the permissions to do this (kafka ACL levels) but we'll call the method anyway which may result in unexpected exceptions
- this makes the mono flow more complicated and unreadable with extracted chains

Reply:
You mentioned an important point. I tried to handle this in 304c880774985d290d412375e196d9ce72fa9015, but I don't like the current state of my code. I would prefer a lambda-like anonymous method instead of `getBrokerConfigIml`. Have you any idea how can I do that?

Reply:
I belive such a logic shouldn't be a part of the controller. 
I suggest the following:
```patch
Subject: [PATCH] Suggestion
---
Index: api/src/main/java/io/kafbat/ui/service/BrokerService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/api/src/main/java/io/kafbat/ui/service/BrokerService.java b/api/src/main/java/io/kafbat/ui/service/BrokerService.java
--- a/api/src/main/java/io/kafbat/ui/service/BrokerService.java	(revision 304c880774985d290d412375e196d9ce72fa9015)
+++ b/api/src/main/java/io/kafbat/ui/service/BrokerService.java	(date 1727477557125)
@@ -59,7 +59,7 @@
     }
     return loadBrokersConfig(cluster, brokerId)
         .map(list -> list.stream()
-            .map(InternalBrokerConfig::from)
+            .map(configEntry -> InternalBrokerConfig.from(configEntry, cluster.isReadOnly()))
             .collect(Collectors.toList()))
         .flatMapMany(Flux::fromIterable);
   }
Index: api/src/main/java/io/kafbat/ui/model/InternalBrokerConfig.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/api/src/main/java/io/kafbat/ui/model/InternalBrokerConfig.java b/api/src/main/java/io/kafbat/ui/model/InternalBrokerConfig.java
--- a/api/src/main/java/io/kafbat/ui/model/InternalBrokerConfig.java	(revision 304c880774985d290d412375e196d9ce72fa9015)
+++ b/api/src/main/java/io/kafbat/ui/model/InternalBrokerConfig.java	(date 1727477557130)
@@ -16,12 +16,12 @@
   private final boolean isReadOnly;
   private final List<ConfigEntry.ConfigSynonym> synonyms;
 
-  public static InternalBrokerConfig from(ConfigEntry configEntry) {
+  public static InternalBrokerConfig from(ConfigEntry configEntry, boolean readOnlyCluster) {
     InternalBrokerConfig.InternalBrokerConfigBuilder builder = InternalBrokerConfig.builder()
         .name(configEntry.name())
         .value(configEntry.value())
         .source(configEntry.source())
-        .isReadOnly(configEntry.isReadOnly())
+        .isReadOnly(readOnlyCluster || configEntry.isReadOnly())
         .isSensitive(configEntry.isSensitive())
         .synonyms(configEntry.synonyms());
     return builder.build();

```

What do you think?

Reply:
Very good idea.

I dropped away my old codes in the controller, applied your suggested changes, and everything seems to work fine.
(I also rebased the branch with main.)

Reply:
Thanks for putting a co-authoring for the commit :-)